[2023-10-27T10:17:50] INFO [root.run:226] Starting model generation process...
[2023-10-27T10:17:50] INFO [root.query_stationary_test_results:68] Querying stationary test results...
[2023-10-27T10:17:50] INFO [root.filter_stationary_test_results:74] Filtering stationary test results...
[2023-10-27T10:17:50] INFO [root.query_top_price_data:89] Querying top price data...
[2023-10-27T10:17:50] INFO [root.generate_crypto_pairs:96] Generating cryptocurrency pairs...
[2023-10-27T10:17:50] INFO [root.create_pivot_table:108] Creating pivot table...
[2023-10-27T10:17:50] INFO [root.generate_iter_list_parallel:204] Generating iter_list in parallel...
[2023-10-27T10:17:51] INFO [root.calculate_residuals:118] Calculating residuals...
[2023-10-27T10:17:52] INFO [root.run_stationary_tests:128] Running stationary tests...
[2023-10-27T10:19:06] INFO [root.filter_stationary_tests_resid:134] Filtering stationary test results on residuals...
[2023-10-27T10:19:06] INFO [root.extract_models_to_make:162] Extracting models to make...
[2023-10-27T10:19:06] INFO [root.generate_linear_regression_models:170] Generating linear regression models...
[2023-10-27T10:19:11] INFO [root.concatenate_models:181] Concatenating linear regression models...
[2023-10-27T10:19:11] INFO [root.save_models_to_database:196] Saving linear regression models to database...
[2023-10-27T10:19:11] ERROR [root.run:243] An unexpected error occurred: table tblModels has no column named training_from_date
Traceback (most recent call last):
  File "<ipython-input-5-ff00a3677994>", line 239, in run
    self.save_models_to_database()
  File "<ipython-input-5-ff00a3677994>", line 199, in save_models_to_database
    self.df_models.to_sql("tblModels", conn, if_exists="append", index=False)
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/core/generic.py", line 3008, in to_sql
    return sql.to_sql(
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/io/sql.py", line 788, in to_sql
    return pandas_sql.to_sql(
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/io/sql.py", line 2439, in to_sql
    return table.insert(chunksize, method)
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/io/sql.py", line 1059, in insert
    num_inserted = exec_insert(conn, keys, chunk_iter)
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/io/sql.py", line 2135, in _execute_insert
    conn.executemany(self.insert_statement(num_rows=1), data_list)
sqlite3.OperationalError: table tblModels has no column named training_from_date
[2023-10-27T10:21:13] INFO [root.run:226] Starting model generation process...
[2023-10-27T10:21:13] INFO [root.query_stationary_test_results:68] Querying stationary test results...
[2023-10-27T10:21:13] INFO [root.filter_stationary_test_results:74] Filtering stationary test results...
[2023-10-27T10:21:13] INFO [root.query_top_price_data:89] Querying top price data...
[2023-10-27T10:21:13] INFO [root.generate_crypto_pairs:96] Generating cryptocurrency pairs...
[2023-10-27T10:21:13] INFO [root.create_pivot_table:108] Creating pivot table...
[2023-10-27T10:21:13] INFO [root.generate_iter_list_parallel:204] Generating iter_list in parallel...
[2023-10-27T10:21:13] INFO [root.calculate_residuals:118] Calculating residuals...
[2023-10-27T10:21:15] INFO [root.run_stationary_tests:128] Running stationary tests...
[2023-10-27T10:22:32] INFO [root.filter_stationary_tests_resid:134] Filtering stationary test results on residuals...
[2023-10-27T10:22:32] INFO [root.extract_models_to_make:162] Extracting models to make...
[2023-10-27T10:22:32] INFO [root.generate_linear_regression_models:170] Generating linear regression models...
[2023-10-27T10:22:37] INFO [root.concatenate_models:181] Concatenating linear regression models...
[2023-10-27T10:22:37] INFO [root.save_models_to_database:196] Saving linear regression models to database...
[2023-10-27T10:22:37] ERROR [root.run:243] An unexpected error occurred: table tblModels has no column named model_coeff
Traceback (most recent call last):
  File "<ipython-input-6-ff00a3677994>", line 239, in run
    self.save_models_to_database()
  File "<ipython-input-6-ff00a3677994>", line 199, in save_models_to_database
    self.df_models.to_sql("tblModels", conn, if_exists="append", index=False)
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/core/generic.py", line 3008, in to_sql
    return sql.to_sql(
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/io/sql.py", line 788, in to_sql
    return pandas_sql.to_sql(
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/io/sql.py", line 2439, in to_sql
    return table.insert(chunksize, method)
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/io/sql.py", line 1059, in insert
    num_inserted = exec_insert(conn, keys, chunk_iter)
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/io/sql.py", line 2135, in _execute_insert
    conn.executemany(self.insert_statement(num_rows=1), data_list)
sqlite3.OperationalError: table tblModels has no column named model_coeff
[2023-10-27T10:23:05] INFO [root.run:226] Starting model generation process...
[2023-10-27T10:23:05] INFO [root.query_stationary_test_results:68] Querying stationary test results...
[2023-10-27T10:23:05] INFO [root.filter_stationary_test_results:74] Filtering stationary test results...
[2023-10-27T10:23:05] INFO [root.query_top_price_data:89] Querying top price data...
[2023-10-27T10:23:06] INFO [root.generate_crypto_pairs:96] Generating cryptocurrency pairs...
[2023-10-27T10:23:06] INFO [root.create_pivot_table:108] Creating pivot table...
[2023-10-27T10:23:06] INFO [root.generate_iter_list_parallel:204] Generating iter_list in parallel...
[2023-10-27T10:23:06] INFO [root.calculate_residuals:118] Calculating residuals...
[2023-10-27T10:23:07] INFO [root.run_stationary_tests:128] Running stationary tests...
[2023-10-27T10:24:29] INFO [root.filter_stationary_tests_resid:134] Filtering stationary test results on residuals...
[2023-10-27T10:24:29] INFO [root.extract_models_to_make:162] Extracting models to make...
[2023-10-27T10:24:29] INFO [root.generate_linear_regression_models:170] Generating linear regression models...
[2023-10-27T10:24:34] INFO [root.concatenate_models:181] Concatenating linear regression models...
[2023-10-27T10:24:34] INFO [root.save_models_to_database:196] Saving linear regression models to database...
[2023-10-27T10:24:34] ERROR [root.run:243] An unexpected error occurred: table tblModels has no column named model_date
Traceback (most recent call last):
  File "<ipython-input-7-ff00a3677994>", line 239, in run
    self.save_models_to_database()
  File "<ipython-input-7-ff00a3677994>", line 199, in save_models_to_database
    self.df_models.to_sql("tblModels", conn, if_exists="append", index=False)
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/core/generic.py", line 3008, in to_sql
    return sql.to_sql(
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/io/sql.py", line 788, in to_sql
    return pandas_sql.to_sql(
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/io/sql.py", line 2439, in to_sql
    return table.insert(chunksize, method)
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/io/sql.py", line 1059, in insert
    num_inserted = exec_insert(conn, keys, chunk_iter)
  File "/Users/simen/anaconda3/envs/crypto-cointegration/lib/python3.10/site-packages/pandas/io/sql.py", line 2135, in _execute_insert
    conn.executemany(self.insert_statement(num_rows=1), data_list)
sqlite3.OperationalError: table tblModels has no column named model_date
[2023-10-27T10:25:13] INFO [root.run:226] Starting model generation process...
[2023-10-27T10:25:13] INFO [root.query_stationary_test_results:68] Querying stationary test results...
[2023-10-27T10:25:13] INFO [root.filter_stationary_test_results:74] Filtering stationary test results...
[2023-10-27T10:25:13] INFO [root.query_top_price_data:89] Querying top price data...
[2023-10-27T10:25:14] INFO [root.generate_crypto_pairs:96] Generating cryptocurrency pairs...
[2023-10-27T10:25:14] INFO [root.create_pivot_table:108] Creating pivot table...
[2023-10-27T10:25:14] INFO [root.generate_iter_list_parallel:204] Generating iter_list in parallel...
[2023-10-27T10:25:14] INFO [root.calculate_residuals:118] Calculating residuals...
[2023-10-27T10:25:15] INFO [root.run_stationary_tests:128] Running stationary tests...
[2023-10-27T10:26:19] INFO [root.filter_stationary_tests_resid:134] Filtering stationary test results on residuals...
[2023-10-27T10:26:19] INFO [root.extract_models_to_make:162] Extracting models to make...
[2023-10-27T10:26:19] INFO [root.generate_linear_regression_models:170] Generating linear regression models...
[2023-10-27T10:26:24] INFO [root.concatenate_models:181] Concatenating linear regression models...
[2023-10-27T10:26:24] INFO [root.save_models_to_database:196] Saving linear regression models to database...
